frontend/ui/network/manager.lua

```diff
@@ -60,6 +60,15 @@ function NetworkMgr:promptWifiOff(complete_callback)
     })
 end
 
+function NetworkMgr:isConnected()
+    if Device:isAndroid() then
+        return self:isWifiOn()
+    else
+        -- `-c1` try only once; `-w2` wait 2 seconds
+        return os.execute("ping -c1 -w2 $(ip route | grep '^default' | awk '/eth0/ {print $3}')") == 0
```

Comment:
I'm not sure if `eth0` is a good idea. On my laptop, the network card is `wls3`. Maybe something like that?
`return os.execute("ping -c1 -w2 $(ip route | grep '^default' | awk '/default/ {print $3}')") == 0`

Reply:
The question is more what it is on your Kindle than what it is on your laptop. :-P

But yes, my original proposal was `ping -c1 -w2 $(ip route | awk '/default/ { print $3 }')` which should be fine.

Reply:
On my Kindle is `wlan0` :)